.changeset/large-mirrors-trade.md

```diff
@@ -0,0 +1,5 @@
+---
+"@finos/legend-application": patch
+"@finos/legend-query": patch
+---
+Move `ValueSpecificationEditor` from `legend-query` to `legend-application`.
```

Comment:
Let's split this changeset, since `QueryBuilderValueSpecification` was never exported from `@finos/legend-query`, we can leave that as `patch` in its own file, without any message.

We should have another changeset for `@finos/legend-application` with `minor` bump since this is a new feature we expose